src/tests/networking_tests.rs

```diff
@@ -1544,6 +1556,167 @@ pub mod net_tests {
+        let cage2 = cage.clone();
+
+        let thread = interface::helper_thread(move || {
+            let mut buf = sizecbuf(10);
+            let mut counter = 0;
+
+            // receive the message
+            // since peer will sleep for 30ms before send the message
+            // some nonblocing returns are expected
+            loop {
+                let result = cage2.recv_syscall(socketpair.sock2, buf.as_mut_ptr(), 10, 0);
+                if result == -(Errno::EAGAIN as i32) {
+                    // return due to nonblocking flag
+                    counter += 1;
+                    continue;
+                }
+                if result != -libc::EINTR {
+                    break; // if the error was EINTR, retry the syscall
+                }
+            }
+            // check if the received message is correct
+            assert_eq!(cbuf2str(&buf), "test\0\0\0\0\0\0");
+            // check if there is any nonblocking return
+            assert_ne!(counter, 0);
+
+            // sleep for 30ms so receiver could have some nonblocking return 
+            interface::sleep(interface::RustDuration::from_millis(30));
+            // send the message
+            assert_eq!(
+                cage2.send_syscall(socketpair.sock2, str2cbuf("Socketpair Test"), 15, 0),
+                15
+            );
+        });
+
+        let cage3 = cage.clone();
+
+        let thread_2 = interface::helper_thread(move || {
+            // sleep for 30ms so receiver could have some nonblocking return
+            interface::sleep(interface::RustDuration::from_millis(30));
+            
+            // send message
+            assert_eq!(cage3.send_syscall(socketpair.sock1, str2cbuf("test"), 4, 0), 4);
+    
+            // receive the message
+            // since peer will sleep for 30ms before send the message
+            // some nonblocing returns are expected
+            let mut buf2 = sizecbuf(15);
+            let mut counter = 0;
+            loop {
+                let result = cage3.recv_syscall(socketpair.sock1, buf2.as_mut_ptr(), 15, 0);
+                if result == -(Errno::EAGAIN as i32) {
+                    // return due to nonblocking flag
+                    counter += 1;
+                    continue;
+                }
+                if result != -libc::EINTR {
+                    break; // if the error was EINTR, retry the syscall
+                }
+            }
+            // check if the received message is correct
+            let str2 = cbuf2str(&buf2);
+            assert_eq!(str2, "Socketpair Test");
+            // check if there is any nonblocking return
+            assert_ne!(counter, 0);
+        });
+
+        thread.join().unwrap();
+        thread_2.join().unwrap();
+
+        assert_eq!(cage.close_syscall(socketpair.sock1), 0);
+        assert_eq!(cage.close_syscall(socketpair.sock2), 0);
+
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
```

Comment:
I learned a lot from reading through your tests! Great work overall!

- Just a note, I would appreciate more comments especially at the top of blocks of code that perform something. This would help me out a lot.